### ISSUE TITLE
Support the AWS' RestJson protocol

### ISSUE THREAD

**Issue by Baccata** [disneystreaming/smithy4s#53]
The AWS RestJson protocol is by AWS to build tooling that is not necessarily related to AWS-SDK . For instance, [smithy-rs](https://github.com/awslabs/smithy-rs) aims at allowing users to derive server-side bindings. 

We should strive to offer client/server that'd respect these protocols, so that we can have compatibility with the tooling that might be created in the larger smithy ecosystem.

**Comment by kubukoz** [comment]
I think we should look at this some time soon, having compatibility with smithy-rs would be useful to have at this point.

**Comment by Baccata** [comment]
> I think we should look at this some time soon 

Thanks for volunteering 😄

**Comment by kubukoz** [comment]
closing in favor of #679

**Comment by kubukoz** [comment]
actually no, I'll just reference it there.

**Comment by Baccata** [comment]
Yeah it's a different thing : this ticket has the server-side handling in in scope, the other one only has client-side with the added constraints of signing, URL setting, etc.
